spec/rspec/core/reporter_spec.rb

```diff
@@ -36,6 +35,10 @@ module RSpec::Core
         reporter.finish
       end
 
+      it "will allow profiling when activated after formatter is used" do
+        allow(config).to receive(:profile_examples?) { true }
+        reporter.finish
+      end
```

Comment:
I still find this spec confusing.

* Why does `allow(config).to receive(:profile_examples?) { true }` count as activating the formatter?
* The doc string says "after formatter is used" but I don't see where the formatter is used...unless `reporter.finish` counts as the formatter being used, but in that case the spec is showing the profiling activated before the formatter is used, in contradiction to the doc string.
* It's not clear what this example is asserting, as there's no expectation.  I know that the original failure was an exception which must not be raised anymore, but without the context of knowing about the original failure I think this spec will be confusing when we come to it in the future.

I suspect that all these issues stem from the fact that this is a highly isolated unit test.  A higher level spec in `spec/integration` might solve these issues.  Or maybe you can find a better way to do this as a unit test :).

Reply:
> * Why does `allow(config).to receive(:profile_examples?) { true }` count as activating the formatter?

We're not activating the formatter, in fact the opposite, we're trying to show that the profiler causes no issues when not initialised manually.

> * The doc string says "after formatter is used" but I don't see where the formatter is used...unless `reporter.finish` counts as the formatter being used, but in that case the spec is showing the profiling activated before the formatter is used, in contradiction to the doc string.

> * It's not clear what this example is asserting, as there's no expectation.  I know that the original failure was an exception which must not be raised anymore, but without the context of knowing about the original failure I think this spec will be confusing when we come to it in the future.

These two points I've tried to address with re-wording, the point is to prove the profiler is now automatically setup.

> A higher level spec in `spec/integration` might solve these issues.  Or maybe you can find a better way to do this as a unit test :).

I tried to use your example to do just that, however it proved non trivial, as we cannot truely replicate this scenario in isolation.